cloudwatch_monitoring/lambdas.py

```diff
@@ -0,0 +1,194 @@
+                        is_iow_lambda = True
+
+    return is_iow_lambda
+
+
+def create_lambda_widgets(region, deploy_stage):
+    """
+    Iterate over an account's list of lambdas and create generic widgets for those with
+    wma:organization = 'IOW' tags.  It also creates some custom widgets.
+
+    :return: List of lambda widgets
+    :rtype: list
+    """
+
+    # set starting and default values for lambda widget positioning and dimensions
+    x, y = [0, 0]
+    lambda_widget_width, lambda_widget_height = [24, 3]
+    lambda_widget_max_width = 24
+
+    lambda_widgets = []
+
+    # grab all the lambdas in the account/region
+    all_lambda_metadata_response = get_all_lambda_metadata(region)
+
+    # iterate over the list of lambda metadata and create widgets for the assets we care about based on filters
+    for function in all_lambda_metadata_response['Functions']:
+
+        if is_iow_asset_filter(function, deploy_stage, region):
+            function_name = function['FunctionName']
+
+            widget = {
+                'type': 'metric',
+                'x': x,
+                'y': y,
+                'height': lambda_widget_height,
+                'width': lambda_widget_width,
+                'properties': {
+                    "metrics": [
+                        ["AWS/Lambda", "ConcurrentExecutions", "FunctionName", function_name],
+                        [".", "Invocations", ".", ".", {"stat": "Sum"}],
+                        [".", "Duration", ".", "."],
+                        [".", "Errors", ".", ".", {"stat": "Sum"}],
+                        [".", "Throttles", ".", "."]
+                    ],
+                    "view": "singleValue",
+                    "region": region,
+                    "title": function_name,
+                    "period": 300,
+                    "stacked": False,
+                    "stat": "Average"
+                }
+            }
+
+            lambda_widgets.append(widget)
+
+            # iterate the position on the dashboard for the next widget
+            x += lambda_widget_width
+            if x + lambda_widget_width > lambda_widget_max_width:
+                x = 0
+                y += lambda_widget_height
+
+    # Custom widget for monitoring concurrency of lambdas specifically involved in the ETL
+    concurrent_lambdas = {
+        'type': 'metric',
+        'x': x,
+        'y': y,
+        'height': lambda_widget_height + 3,
+        'width': lambda_widget_width,
+        'properties': {
+            "metrics": [
+                ["AWS/Lambda", "ConcurrentExecutions", "FunctionName", "aqts-capture-dvstat-transform-" + deploy_stage + "-transform", {"label": "DV stat Transformer"}],
+                ["...", "aqts-capture-error-handler-" + deploy_stage + "-aqtsErrorHandler", {"label": "Error Handler"}],
+                ["...", "aqts-capture-raw-load-" + deploy_stage + "-iowCapture", {"label": "Raw Loader"}],
```

Comment:
You could also use f-strings instead of old school string concatenation: https://realpython.com/python-f-strings/

```python
f"aqts-capture-raw-load-{deploy_stage}-iowCapture"
```

Reply:
Done.